pkg/agent/agent.go

```diff
@@ -206,7 +165,54 @@ func (k *Klocksmith) Run() error {
 	k.lc.Reboot(false)
 
 	// cross fingers
-	time.Sleep(24 * 7 * time.Hour)
+	sleepOrDone(24*7*time.Hour, stop)
+	return nil
```

Comment:
Hm, what about after 24 hours have passed? sleepOrDone should be just like the time.Sleep call that was there before (followed by reaching the end of the agent behavior, effectively the same as return nil here), except it can be interupted if the top-level stop were closed.